.github/workflows/builder_nodejs_slsa3.yml

```diff
@@ -71,15 +76,30 @@ jobs:
       contents: write # For asset uploads. Optional
     runs-on: ubuntu-latest
     steps:
+      - name: Set up Node environment
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        with:
+          node-version: ${{ inputs.node-version }}
+          node-version-file: ${{ inputs.node-version-file }}
+          registry-url: "https://registry.npmjs.org"
+
+      # NOTE: The attestation name has a rng value in it's name so we don't need to check for file collisions.
       - name: Download attestations
+        id: attestation-download
         uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
         with:
           name: ${{ needs.slsa-run.outputs.attestations-download-name }}
 
-      - name: Download artifact
-        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+      # NOTE: secure-download-artifact ensures that the downloaded file doesn't overwrite an existing file.
+      - name: Download package
+        id: package-download
```

Comment:
so this would mean we'll have to expose the action `secure-download-artifact` to TRWs in the future. Can we instead download insecurely and use the slsa-verifier to verify the artifact is the correct one?

What we'll need is:
1. BYOB support inn slsa-verifier
2. Support for `--source-digest` in slsa-verifier, to prevent someone from replacing the genuine artifact with an older artifact

Wdut?

Reply:
> I think provenance should also be securely downloaded.

This finally only made sense to me since the TRW process should not be influenced by the maintainer, and we need to avoid the potential provenance confusion Laurent mentioned above.

I expect in this case we are only creating 1 attestation file, but the delegator handles a folder of attestations inc ase the layout specified more. The upload:
https://github.com/slsa-framework/slsa-github-generator/blob/ab4b569619e185570320838296f57451d79d3547/.github/workflows/delegator_generic_slsa3.yml#L257

I think it should be fairly easy to support folders in secure upload/download, if we tar the folders before and after download to checksum

Reply:
In fact, I think the provenance is the only file that needs to be securely downloaded. The artifact verification can then be done via slsa-verifier. It would fail if the user changed the artifact. This would probably also simplify the Action. So it would look like this:

1. TRW MUST securely download the attestation we return
2. TRW MUST slsa-verifier with `--source-uri=$GITHUB_REPOSITORY --build-id <their-builder-id>` before uploading artifact

Reply:
> I think it should be fairly easy to support folders in secure upload/download, if we tar the folders before and after download to checksum

That sounds like a good idea. Do we have an issue for it already?

Reply:
Created https://github.com/slsa-framework/slsa-github-generator/issues/1863.

Reply:
Thanks. I think this is a good approach. I think we can address this on a follow up PR.